test/compile.jl

```diff
@@ -474,4 +524,61 @@ let module_name = string("a",randstring())
     rm(file_name)
 end
 
+# Issue #19960
+let
+    # ideally this would test with workers on a remote host that does not have access to the master node filesystem for loading
+    # can simulate this for local workers by using relative load paths on master node that are not valid on workers
+    # so addprocs before changing directory to temp directory, otherwise workers will inherit temp working directory
+
+    test_workers = addprocs(1)
+
+    temp_path = mktempdir()
+    cd(temp_path)
+    load_path = mktempdir(temp_path)
+    load_cache_path = mktempdir(temp_path)
+    unshift!(LOAD_PATH, basename(load_path))
+    unshift!(Base.LOAD_CACHE_PATH, basename(load_cache_path))
+
+    ModuleA = :Issue19960A
+    ModuleB = :Issue19960B
+
+    write(joinpath(load_path, "$ModuleA.jl"),
+        """
+        __precompile__(true)
+        module $ModuleA
+            export f
+            f() = myid()
+        end
+        """)
+
+    write(joinpath(load_path, "$ModuleB.jl"),
+        """
+        __precompile__(true)
+        module $ModuleB
+            using $ModuleA
+            export g
+            g() = f()
+        end
+        """)
+
+    try
+        @eval using $ModuleB
+
+        @test g() == 1
+        for wid in test_workers
+            @test remotecall_fetch(g, wid) == wid
+        end
+
+    catch
+        @test false
```

Comment:
Given your comment, it probably doesn't need to be there.
(As you've probably guessed I don't really understand testing) 
I thought that there had to be an explicit `@test` failure, rather than just a thrown error from the try block.
I can remove it.